Webapp/sources/blueprints/reaction_list/routes.py

```diff
@@ -104,8 +105,11 @@ def get_schemata() -> Response:
 @reaction_list_bp.route("/get_new_reaction_id", methods=["GET", "POST"])
 @login_required
 def get_new_reaction_id() -> Response:
-    workbook_name = request.json['workbook']
-    workgroup_name = request.json['workgroup']
+    workbook_name = request.json.get("workbook")
+    workgroup_name = request.json.get("workgroup")
+
+    if workgroup_name is None or workbook_name is None:
+        return abort(400)
```

Comment:
I noticed you've installed flask api below. You can import `status` from `flask_api` and then use  `status.HTTP_BAD_REQUEST` instead of just `400`. It also looks clearer to people maintaining the code.
https://flask-api.github.io/flask-api/api-guide/status-codes/